src/components/CustomerDetails/CollectCustomerDetailsScreen.tsx

```diff
@@ -173,6 +173,25 @@ const CollectCustomerDetailsScreen: FunctionComponent<NavigationFocusInjectedPro
   }, [selectionArray, selectedIdType, setSelectedIdType]);
 
   const onCheck = async (input: string): Promise<void> => {
+    /**
+     * If "Passport" tab is chosen, we want to extract the passport
+     * number from a JSON object containing the `passportId`.
+     */
+    if (selectedIdType.label === "Passport") {
+      let passportId;
+      try {
+        ({ passportId } = JSON.parse(input));
+      } catch (e) {
+        /**
+         * We set this to empty so an appropriate error dialog will be
+         * shown, instead of an ErrorBoundary or empty dialog.
+         */
```

Comment:
wah interesting, so after leaving this empty, how is the error handled?

Reply:
is it caught by error boundary ah? do we need to throw out the error again after catching it?

Reply:
This error is thrown because it's not a valid JSON object, so there's no `passportId` to parse. We leave it empty so that the error is handled below, similar to the case when there's no input at all.

Reply:
ok understood! in that case, will it be better for us to catch the syntax error (i.e. `catch (e instanceof SyntaxError)` instead of catching all errors, so that it's clearer that this block is use specifically for json parsing errors?

in the event there are any unforeseen errors, it will go to the error boundary instead